server/src/metadata.rs

```diff
@@ -321,11 +321,13 @@ pub mod entities_metadata_query {
                             .collect()
                     })
                     .unwrap_or(Vec::new());
-                let subject_metadata: Option<Vec<SubjectMetadata>> = subject_ids.iter().map(|id| {
-                    let raw_subject_metadata = map_serlo_subjects_to_amb_standard(*id);
-                    raw_subject_metadata.map(|data| data.into())
+                let subject_metadata_raw: Vec<SubjectMetadata> = subject_ids.iter().flat_map(|id| {
+                    map_serlo_subjects_to_amb_standard(*id)
                 }).collect();
-
+                let subject_metadata = match subject_metadata_raw.is_empty() {
```

Comment:
I made https://github.com/serlo/database-layer/pull/483 for an alternative suggestion.